migrator/migrations/m_200_to_m_201_vuln_request_v1_to_v2/migration_impl.go

```diff
@@ -0,0 +1,134 @@
+package m200tom201
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations/m_200_to_m_201_vuln_request_v1_to_v2/schema"
+	"github.com/stackrox/rox/migrator/migrations/m_200_to_m_201_vuln_request_v1_to_v2/store"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+var (
+	batchSize = 2000
+	log       = logging.LoggerForModule()
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	pgutils.CreateTableFromModel(ctx, database.GormDB, schema.CreateTableVulnerabilityRequestsStmt)
+
+	return updateVulnerabilityRequests(ctx, database)
+}
+
+func updateVulnerabilityRequests(ctx context.Context, database *types.Databases) error {
+	vulnReqStore := store.New(database.PostgresDB)
+
+	var updatedObjs []*storage.VulnerabilityRequest
+	var count int
+	err := vulnReqStore.Walk(ctx, func(obj *storage.VulnerabilityRequest) error {
+
+		updated := obj.Clone()
+
+		// Migrate requester and approvers to new RequesterV2 and ApproversV2 fields
+		if requester := obj.GetRequestor(); requester != nil {
+			updated.RequesterV2 = convertSlimUserToRequester(obj.GetRequestor())
+		}
+		if approvers := obj.GetApprovers(); len(approvers) > 0 {
+			updated.ApproversV2 = convertSlimUserToApprovers(obj.GetApprovers())
+		}
+
+		// Update new expiry type field
+		if obj.GetDeferralReq() != nil && obj.GetDeferralReq().GetExpiry() != nil {
+			if obj.GetDeferralReq().GetExpiry().GetExpiresWhenFixed() {
+				updated.GetDeferralReq().Expiry.ExpiryType = storage.RequestExpiry_ANY_CVE_FIXABLE
+			} else {
+				updated.GetDeferralReq().Expiry.ExpiryType = storage.RequestExpiry_TIME
+			}
+		}
+
+		// Migrate UpdateDeferralReq (if any) to new DeferralUpdate field
+		if obj.GetUpdatedDeferralReq().GetExpiry() != nil {
+			updated.UpdatedReq = &storage.VulnerabilityRequest_DeferralUpdate{
+				DeferralUpdate: &storage.DeferralUpdate{
+					CVEs:   obj.GetCves().GetCves(),
```

Comment:
Not obvious what the type of `obj.GetCves()` is or it it is possible to be `nil`, but do we need to protect from it being `nil`?

Reply:
It shouldn't ideally be nil here because the validations at the service level do not allow that when deferral requests are posted. But looking at the migration independently, even if `obj.GetCves()` is nil, `obj.GetCves().GetCves()` will return nil too as both `GetCves()` funcs do a nil check. So there won't be any crash.

Reply:
Perfect.